### DISCUSSION TITLE
Do preview deployments use API scripts from production branch/deployment?

### DISCUSSION THREAD

**Discussion by cezarsmpio** [vercel/vercel#5650] [Help]
Hey everyone, I have a pertinent question, not sure if this is the correct behavior or a bug, but I was testing a preview deploy, the one that Vercel generates automagically when a Pull Request is created, however, my `/api` endpoints are kinda using the results from the production `/api` endpoints.

To give you more context, on a new branch, I changed one of my API endpoint response, and when I was testing the preview deploy, I was getting the response from the production version, not the one I've just changed for the preview.

My project uses Next.js with the `pages/api` folder for lambda functions.

Thanks!

**Comment by styfle** [comment]
Hi @cezarsmpio That is not expected.

Perhaps you have the same environment variables for Production and Preview which is why you would get the same results (assumed your API is talking to a database).

https://vercel.com/docs/environment-variables

**Reply by cezarsmpio** [reply]
Hi @styfle , the problem was not the result but the format of the response. Let's say in production I'm returning this format:

Production
```js
{ data: [{ name: "Cezar" }] }
```

But in my branch I made some changes to the format, so I started returning this format instead:

Branch
```js
{ data: { users: [{ name: "Cezar" }] } }
```

When I was testing my branch on the preview deployment, I was getting the format from production.

**Reply by styfle** [reply]
That doesn't sound right 🤔 

Please [create a support ticket](https://vercel.com/support/request) so we can look at your account with the deployment in question.

Thanks!
